Extension/src/LanguageServer/configurations.ts

```diff
@@ -2050,13 +2071,33 @@ export class CppProperties {
                     if (curOffset >= compilerPathStart && curOffset <= compilerPathEnd) {
                         continue;
                     }
-                    let message: string;
+                    let message: string = "";
                     if (!pathExists) {
                         if (curOffset >= forcedIncludeStart && curOffset <= forcedeIncludeEnd
                                 && !path.isAbsolute(resolvedPath)) {
                             continue; // Skip the error, because it could be resolved recursively.
                         }
-                        message = localize('cannot.find2', "Cannot find \"{0}\".", resolvedPath);
+                        // If there are incorrect paths and the first match includes "env:", split the string and check for environment variable.
+                        if (incorrectExpandedPaths.length > 0 && configMatches[0].includes("env:")) {
+                            const splitString = configMatches[0].split("env:");
+                            if (splitString[1]) {
+                                const matchResult = splitString[1].match(/\w+/);
+                                if (matchResult) {
+                                    const envVar = matchResult[0];
+                                    message = localize('cannot.find3', "Cannot find {0} in environment variable: {1}.", incorrectExpandedPaths.map(s => `"${s}"`).join(', '), envVar);
+                                }
+                            }
+                        // If there are incorrect paths but no "env:" in the first match, generate a message indicating the paths cannot be found.
+                        } else {
+                            let badPath = "";
+                            if (incorrectExpandedPaths.length > 0) {
+                                badPath = incorrectExpandedPaths.map(s => `"${s}"`).join(', ');
+                            } else {
+                                badPath = `"${resolvedPath}"`;
```

Comment:
I don't think `resolvedPath` is a needed variable anymore.  `resolveAndSplit` and the loop on the `expandedPaths` that it returns should be doing all the stuff that we used to do with `resolvedPath` instead. It feels like there are two distinct, competing implementations in these 100 lines of code.